riyaz/db.py

```diff
@@ -289,6 +308,50 @@ def set(cls, key, value):
         return kv.save()
 
 
+class Asset(Document):
+    _TABLE = "asset"
+
+    collection: str
+    collection_id: int
+    filename: str
+
+    filesize: Optional[int]
+    created: Optional[datetime]
+    last_modified: Optional[datetime]
+
+    def _get_full_identifier(self):
+        return f"{self.collection}/{self.collection_id}/{self.filename}"
+
+    def _construct_asset_path(self):
+        assert "/" not in self.filename, "filename should not be a path"
+
+        full_identifier = self._get_full_identifier()
+        return Path(config.assets_path) / full_identifier
+
+    def get_url(self):
+        assets_root = "/assets/"
+
+        full_identifier = self._get_full_identifier()
+        return f"{assets_root}{full_identifier}"
+
+    def save_file(self, on_disk_path: Path):
+        assert on_disk_path.is_file()
```

Comment:
It may be a good idea to check and raise `ValueError` if the path is not present rather than using `assert`.

Reply:
Done.